Modules/_hashopenssl.c

```diff
@@ -538,19 +541,178 @@ static PyType_Spec EVPtype_spec = {
     EVPtype_slots
 };
 
+#ifdef PY_OPENSSL_HAS_SHAKE
+
+/*[clinic input]
+_hashlib.HASHXOF.digest as EVPXOF_digest
+
+  length: size_t
+
+Return the digest value as a bytes object.
+[clinic start generated code]*/
+
+static PyObject *
+EVPXOF_digest_impl(EVPobject *self, size_t length)
```

Comment:
Use Py_ssize_t instead of size_t.  the API it is being used on is constrained by Py_ssize_t limits.  It is practical for a 32-bit platform with 4G ram to bump into the difference.